frontend/src/app/modules/user-profile/user-details/user-details/user-details.component.ts

```diff
@@ -102,15 +125,23 @@ export class UserDetailsComponent extends BaseComponent implements OnInit {
         return this.detailsForm.get('email') as FormControl;
     }
 
-    get instagram(): FormControl {
-        return this.detailsForm.get('instagram') as FormControl;
+    selectInterest($event: Event, i: number) {
+        const ev = $event.target as HTMLInputElement;
+
+        this.selectedTags = (ev.checked)
+            ? [...this.selectedTags, { name: this.tagsList[i].icon_name } as ITag]
```

Comment:
find not by index, find by Id

Reply:
Interface `ITag` doesn't have `Id`. And for now we get images for Tags on FE, so name is a reliable key for finding the image. Actually, it will be changed later to used DB for loading imges as well. Pls clarify)
`export interface ITag {
    name: string;
}`

Reply:
we have Tag entity in the backend, this means that all our tags should have an Id

Reply:
fixed